### PR TITLE
Added `possible_values()`

### PR DESCRIPTION
**Pull request by franramirez688**
Docs for https://github.com/conan-io/conan/pull/13796

### CODE REVIEW

File: reference/conanfile/attributes/binary_model.inc [hunk 1]
```diff
@@ -74,9 +74,27 @@ If you want to do a safe check of settings values, you could use the ``get_safe(
...
+    def generate(self):
+        definition = self.settings.compiler.version.possible_values()
+        self.output.info("HOST: " + ", ".join(definition))
+        definition = self.settings_build.compiler.version.possible_values()
+        self.output.info("BUILD: " + ", ".join(definition))
+        cppstds = self.settings.compiler.cppstd.possible_values()
+        self.output.info("CPPSTDS: " + str(cppstds))
+
+
+This method is layer-based so doing ``self.settings.possible_values()`` returns the whole :ref:`reference_config_files_settings_yml`
```
Comment (czoido) [inline reference/conanfile/attributes/binary_model.inc:94]: I think we should better not use that 'layer-based' expression and maybe extending the example on how it works a bit

Comment (RubenRBS) [reply reference/conanfile/attributes/binary_model.inc:94]: Agree

**Review by RubenRBS — APPROVED** [review]

File: reference/conanfile/attributes/binary_model.inc (hunk 1 above)
Comment (franramirez688) [reply reference/conanfile/attributes/binary_model.inc]: Now, it's explained according to the example above.

File: reference/conanfile/attributes/binary_model.inc [hunk 2]
```diff
@@ -74,9 +74,31 @@ If you want to do a safe check of settings values, you could use the ``get_safe(
...
+        # Print the available compiler versions given your host profile
+        compiler_versions = self.settings.compiler.version.possible_values()
+        self.output.info(f"[HOST] Versions for {str(self.settings.compiler)}:  {', '.join(compiler_versions)}")
+        # Print the available compiler versions given your build profile
+        compiler_versions = self.settings_build.compiler.version.possible_values()
+        self.output.info(f"[BUILD] Versions for {str(self.settings.compiler)}:  {', '.join(compiler_versions)}")
+
+
+As you can see above, this method is layer-based so doing ``self.settings.possible_values()`` returns the
+whole :ref:`reference_config_files_settings_yml` as a Python dict-like object, and doing
```
Comment (czoido) [inline reference/conanfile/attributes/binary_model.inc:98]:
```suggestion
As you can see above, doing ``self.settings.possible_values()`` returns the
whole :ref:`reference_config_files_settings_yml` as a Python dict-like object, and doing
```

File: reference/conanfile/attributes/binary_model.inc [hunk 3]
```diff
@@ -74,9 +74,31 @@ If you want to do a safe check of settings values, you could use the ``get_safe(
         # Will be the default version if the return is None
         build_type = self.settings.get_safe("build_type", default="Release")
 
-The ``get_safe()`` method will return ``None`` if that setting or sub-setting doesn't
+The ``get_safe()`` method returns ``None`` if that setting or sub-setting doesn't
```
Comment (czoido) [inline reference/conanfile/attributes/binary_model.inc:77]: What about adding some link here in the introduction to the settings.yml and mentioning that these values come from that file?

**Review by czoido — APPROVED** [review]

**Pull request merged**